### PR TITLE
Redesign Home screen

### PR DESCRIPTION
**Pull request by StylianosGakis**
Todo:
* Handle the PSA according to whatever the new design will look like.
* Create the layout which will try to center the content in the existing screen space instead of it being in a simple column
* Look into the font used in everything since some things feel much bigger than others
* The claim's card also needs some attention, it was never changed from whatever we already had there already. But I think you've gone over this for the claim detail screen already
* The chat reminder bubble at the top right logic of when it shows is handled in the composable UI right now by accessing sharedpreferences, that's definitely not optimal, but it's just how it was before and I didn't touch it here.
* Determine when the top right icons in Home should show. Particularly find when the FirstVet icon should be showing, maybe filter existing contracts and see if any is of type pet? Also not sure when the green one shows or where it goes when clicking it.
* Change the chat icon on the top right to use the new design

image

image

### CODE REVIEW

**Comment by hugokallstrom** [comment]
"Determine when the top right icons in Home should show. Particularly find when the FirstVet icon should be showing, maybe filter existing contracts and see if any is of type pet? Also not sure when the green one shows or where it goes when clicking it."

@julia-andersson how is the logic on iOS here for showing firstVet icon? Also can you post a screenshot of the "cross sell bottomsheet" that we should show when pressing the green icon?

File: app/feature/feature-home/src/main/kotlin/com/hedvig/android/feature/home/home/data/GetHomeDataUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+          languageService.getGraphQLLocale(),
+          languageService.getGraphQLLocale().rawValue,
+        ),
+      )
+        .fetchPolicy(if (forceNetworkFetch) FetchPolicy.NetworkOnly else FetchPolicy.CacheFirst)
+        .safeFlow(::ErrorMessage),
+      getMemberRemindersUseCase.invoke(),
+      flow { emit(getTravelCertificateSpecificationsUseCase.invoke().getOrNull()) },
+    ) { homeQueryDataResult, memberReminders, travelCertificateData ->
+      either {
```
Comment (hugokallstrom) [inline app/feature/feature-home/src/main/kotlin/com/hedvig/android/feature/home/home/data/GetHomeDataUseCase.kt:53]: Can we log the error message here?

**Review by hugokallstrom — APPROVED** [review]
Great work!

File: app/feature/feature-home/src/main/kotlin/com/hedvig/android/feature/home/home/ui/HomePresenter.kt [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+          allowGeneratingTravelCertificate = successData.allowGeneratingTravelCertificate,
+          emergencyData = successData.emergencyData,
+        )
+      }
+    }
+  }
+}
+
+internal sealed interface HomeEvent {
+  object RefreshData : HomeEvent
```
[the remark this answers is not in the document]
Comment (StylianosGakis) [reply app/feature/feature-home/src/main/kotlin/com/hedvig/android/feature/home/home/ui/HomePresenter.kt:92]: Done ✅
Gotta get used to it

File: app/feature/feature-home/src/main/kotlin/com/hedvig/android/feature/home/home/data/GetHomeDataUseCase.kt (hunk 1 above)
Comment (StylianosGakis) [reply app/feature/feature-home/src/main/kotlin/com/hedvig/android/feature/home/home/data/GetHomeDataUseCase.kt:53]: yep ✅

File: app/feature/feature-home/src/main/kotlin/com/hedvig/android/feature/home/home/ui/HomePresenter.kt [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+) : MoleculePresenter<HomeEvent, HomeUiState> {
+  @Composable
+  override fun MoleculePresenterScope<HomeEvent>.present(lastState: HomeUiState): HomeUiState {
+    var hasError by remember { mutableStateOf(false) }
+    var isLoading by remember { mutableStateOf(false) }
+    var successData: SuccessData? by remember { mutableStateOf(SuccessData.fromLastState(lastState)) }
+    var loadIteration by remember { mutableStateOf(0) }
+
+    CollectEvents { homeEvent: HomeEvent ->
+      when (homeEvent) {
```
[the remark this answers is not in the document]
Comment (StylianosGakis) [reply app/feature/feature-home/src/main/kotlin/com/hedvig/android/feature/home/home/ui/HomePresenter.kt:34]: ✅

**Pull request merged**